### DISCUSSION TITLE
ddev Windows 10 debugging with VSCODE - needs always reboot

### DISCUSSION THREAD

**Discussion by ArminGe** [ddev/ddev#2794] [Q&A]
I have installed ddev on Windows 10 (WSL2). Everything works except debugging with VScode. This always works only when I restart the Windows machine once completely. This is a bit annoying because I can't really explain it. Does anyone have any ideas?

**Comment by rfay** [comment]
Could you please give more details? What only works when you restart Windows? WSL2? vscode? debugging? Are you saying you have to restart your computer to get xdebug to work? 

There's an xdebug troubleshooting section in the docs at https://ddev.readthedocs.io/en/latest/users/step-debugging/#troubleshooting-xdebug - maybe that will help you understand what's going on.

**Reply by ArminGe** [reply]
When I start ddev for the first time, the dddev container works perfectly. xdebug is enabled, debugging in vscode does not work.
If I restart the computer, start the container again, xdebug is enabled, debugging also works.

**Reply by rfay** [reply]
Please  go through the [troubleshooting steps](https://ddev.readthedocs.io/en/latest/users/step-debugging/#troubleshooting-xdebug) when you have the problem. I suppose it could be firewall or network related.

**Reply by ArminGe** [reply]
Coming back to the topic: I start Windows 10 once, ddev works. But I can not access port 9000 via telenet. I restart the computer and then ddev, it works on the second try and I can connetc to Port 9000. What happens that I need two starts?

**Reply by rfay** [reply]
So it's vscode that listens on port 9000. If you're not able to `telnet localhost 9000` on the host (on the windows side) after you've started debugging then something's broken about vscode. If the problem is that you can get to it on the host side but not in the container side, then it's likely firewall or something like that. 

But the bottom line: It's vscode that listens on port 9000, if it's not listening then you won't be able to access it. You get it listening by clicking the run-with-debug triangle.

**Reply by ArminGe** [reply]
Thanks, but all your tips don't really help me. I know that Visual Studio code must listen to the port. And it does. What I don't understand is why it doesn't work directly after starting the comoputer and ddev. It works without changing any settings only when I restart the whole computer, Windows and ddev. With the first start nothing works, after the reboot it works. I do not understand this.

**Reply by rfay** [reply]
You should open an issue in the vscode issue queue if you can recreate this. But it doesn't have to do with ddev. 

I'm sure you do understand that vscode doesn't start listening until you press the debug button right?
